data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/RegexNotEqualOperator.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.expression;
+
+import org.opensearch.dataprepper.expression.antlr.DataPrepperExpressionParser;
+
+import java.util.regex.PatternSyntaxException;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+class RegexNotEqualOperator implements Operator<Boolean> {
+    @Override
+    public Integer getSymbol() {
+        return DataPrepperExpressionParser.NOT_MATCH_REGEX_PATTERN;
+    }
+
+    @Override
+    public Boolean evaluate(Object... args) {
+        checkArgument(args.length == 2, "Operands length needs to be 2.");
+        checkArgument(args[0] instanceof String, "Left operand needs to be String.");
+        checkArgument(args[1] instanceof String, "Right Operand needs to be String.");
+        try {
+            return !((String) args[0]).matches((String) args[1]);
+        } catch (final PatternSyntaxException e) {
+            throw new IllegalArgumentException(e);
+        }
+    }
```

Comment:
Did we consider simply negating the operations to reduce the code and checks? If our regex (or others ) implementation has to change we now have to update multiple classes. Leveraging existing implementations would simplify this and reduce code duplication again.

```
class GenericNotOperator implements Operator<Boolean> {

    private final Integer symbol;
    private final Operator<Boolean> operationToNegate;

    public GenericNotOperator(final Integer symbol, Operator<Boolean> operationToNegate) {
        this.symbol = symbol;
        this.operationToNegate = operationToNegate;
    }

    @Override
    public Integer getSymbol() {
        return this.symbol;
    }

    @Override
    public Boolean evaluate(final Object... args) {
        return !this.operation.evaluate(args);
    }
}
```

Ideally we can inject these through DI.
```
    public Operator<Boolean> notMatchRegexPatternBuilder() {
        return new GenericNotOperator(DataPrepperExpressionParser.NOT_MATCH_REGEX_PATTERN, new RegexEqualOperator());
    }

    public Operator<Boolean> notInBuilder() {
        return new GenericNotOperator(DataPrepperExpressionParser.NOT_IN_PATTERN, new NotInOperator());
    }
```